ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -146,6 +150,12 @@ public static boolean isSystemColumn(final ColumnName columnName) {
             1, VERSION_ONE_NAMES
         );
 
+    private static final Set<ColumnName> MUST_BE_MATERIALIZED_FOR_TABLE_JOINS =
```

Comment:
A slightly cleaner way to structure this would be to create an object representing a pseudocolumn. This object could contain the pseudocolumn name and a boolean to specify whether the column should be materialized or not. By tying the two together in this way, it is not possible for someone to come along and add a new pseudocolumn without thinking about whether that new column should be materialized or not.

That said, I think what you've implemented here is fine for now since adding pseudocolumns to the codebase is rare.

Reply:
This is a really good thing to think about, I haven't really gone "up" a layer in abstraction (ie added my own)- but I've seen it a lot in the codebase, and while it's taken a while to get used to, I can definitely see how it helps maintainable and readability in the long term. I'd be interested in doing this, let me earmark it as something I can do later for time's sake though.